{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2bbd162b",
   "metadata": {},
   "source": [
    "## FASTQC (Análisis de calidad de secuencias)\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a05b8a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -p cicese\n",
    "#SBATCH --job-name=fastqc\n",
    "#SBATCH --output=fastqc-%j.log\n",
    "#SBATCH --error=fastqc-%j.err\n",
    "#SBATCH -N 1\n",
    "#SBATCH --ntasks-per-node=8\n",
    "#SBATCH -t 6-00:00:00\n",
    "\n",
    "\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/FastQC_v0.11.7/:$PATH\n",
    "\n",
    "module load gcc-7.2.0\n",
    "\n",
    "fastqc ../secuencias/*.fastq -t 8 -o ./"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c419cfe3",
   "metadata": {},
   "source": [
    "##### Para realizar multiqc en el cluster se debe colocar el module conda-2023 y la instrucción de multiqc\n",
    "multiqc ../*.zip -o ./ --data-format tsv --export --title \"Analisis pre trimming\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9534020",
   "metadata": {},
   "source": [
    "---\n",
    "Trimmomatic\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6b2d232",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#########################################################\n",
    "#TRIMOMMATIC ##prueba 2\n",
    "\n",
    "## Directivas\n",
    "#SBATCH --job-name=qtrim\n",
    "#SBATCH --output=trimmomatic-%j.log\n",
    "#SBATCH --error=trimmomatic-%j.err\n",
    "#SBATCH -N 1\n",
    "#SBATCH --mem=20GB\n",
    "#SBATCH --ntasks-per-node=8\n",
    "#SBATCH -t 6-00:00:00\n",
    "#SBATCH -p cicese\n",
    "\n",
    "\n",
    "# Ruta a TRIMMOMATIC\n",
    "TRIMMOMATIC=/LUSTRE/apps/bioinformatica/Trimmomatic-0.39\n",
    "\n",
    "#Ruta al archivo con los adaptadores\n",
    "trueseq=\"$TRIMMOMATIC/adapters/TruSeq3-PE-2.fa\"\n",
    "trueseq=\"$TRIMMOMATIC/adapters/TruSeq3-PE.fa\"\n",
    "trueseq=\"$TRIMMOMATIC/adapters/TruSeq2-PE.fa\"\n",
    "\n",
    "cd ${SLURM_SUBMIT_DIR}\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_C1_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_C1_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_C1.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_C2_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_C2_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_C2.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_C3_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_C3_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_C3.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_F1_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_F1_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_F1.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_F2_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_F2_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_F2.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/BLA_F3_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/BLA_F3_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout BLA_F3.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_C1_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_C1_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_C1.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_C2_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_C2_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_C2.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_C3_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_C3_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_C3.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_F1_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_F1_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_F1.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_F2_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_F2_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_F2.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "java -jar $TRIMMOMATIC/trimmomatic-0.39.jar PE ../raw/LOL_F3_CKDL230014589-1A_HJLMVCCX2_L5_1.fq.gz ../raw/LOL_F3_CKDL230014589-1A_HJLMVCCX2_L5_2.fq.gz -baseout LOL_F3.fq.gz \\\n",
    "ILLUMINACLIP:$trueseq:2:30:10:8:true SLIDINGWINDOW:4:15 CROP:110 HEADCROP:10 MINLEN:36\n",
    "\n",
    "exit 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f4959f6",
   "metadata": {},
   "source": [
    "###### Para conocer más sobre los comandos y flags [Manual trimmomatic](http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b68e44e",
   "metadata": {},
   "source": [
    "---\n",
    "## Trinity (Ensamble de novo)\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09f368ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Directivas\n",
    "#SBATCH --job-name=Trinity-denovo # nombre del trabajo que aparecera\n",
    "#SBATCH --output=slurm-%j.log # salida estandar num de trabajo\n",
    "#SBATCH --error=slurm-%j.err # salida estandar de error\n",
    "#SBATCH --nodes 2 # número de nodos\n",
    "#SBATCH --mem=100GB # RAM 1GB por cada millon de reads\n",
    "#SBATCH --ntasks-per-node=24 # número de tareas por nodo\n",
    "#SBATCH -t 6-00:00:00 # número maximo\n",
    "#SBATCH -p cicese #cola\n",
    "\n",
    "\n",
    "#export some soft\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/bowtie2/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/samtools-1.17/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/jellyfish-2.3.0/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/salmon/bin/\n",
    "module load conda-2023\n",
    "module load trinityrnaseq-v2.15.1\n",
    "\n",
    "#my vars\n",
    "\n",
    "PH=`pwd`\n",
    "R1=\"$PH/R1.fastq.gz\"\n",
    "R2=\"$PH/R2.fastq.gz\"\n",
    "OUTDIR=\"Trinity_ensamble\"\n",
    "LIBT=\"RF\"\n",
    "Trinity --no_version_check --seqType fq --max_memory 100G \\\n",
    "--no_bowtie \\\n",
    "--CPU 24 --left $R1 --right $R2 \\\n",
    "--normalize_reads \\\n",
    "--output $OUTDIR\n",
    "#--full_cleanup\"\n",
    "\n",
    "# see the time spent\n",
    "START=`date +%s`\n",
    "\n",
    "# Run trinity\n",
    "###$TRINITY  $PARM\n",
    "\n",
    "END=`date +%s`\n",
    "\n",
    "RUNTIME=$(($END-$START))\n",
    "\n",
    "echo \"run time -> $RUNTIME\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65797297",
   "metadata": {},
   "source": [
    "###### Para conocer más sobre los comandos y flags [Trinity](https://github.com/trinityrnaseq/trinityrnaseq/wiki)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5402f22",
   "metadata": {},
   "source": [
    "#### Una vez que se realiza el análisis, se limpian las lecturas y se realiza el ensamble, se debe revisar la calidad con la que se hizo dicho ensamble y para ello se usan herramientas como transrate o busco"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dee0cb28",
   "metadata": {},
   "source": [
    "---\n",
    "Transrate/Ruby\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fb184c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "#SBATCH -p cicese\n",
    "#SBATCH --job-name=tst-rvm\n",
    "#SBATCH --output=ruby-%j.log\n",
    "#SBATCH --error=ruby-%j.err\n",
    "#SBATCH -N 1\n",
    "#SBATCH --mem=100GB\n",
    "#SBATCH --ntasks-per-node=24\n",
    "#SBATCH -t 06-00:00:00\n",
    "\n",
    "##se deben concatenar todas las trimm_p forward en un archivo y las reverse en otro\n",
    "\n",
    "TRANSRATE=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/bin\n",
    "ENSAMBLE=\"../Trinity.fasta\"\n",
    "R1=\"../R1.fastq.gz\"\n",
    "R2=\"../R2.fastq.gz\"\n",
    "\n",
    "cd $SLURM_SUBMIT_DIR\n",
    "\n",
    "export PATH=/LUSTRE/apps/bioinformatica/.local/bin:$PATH\n",
    "export PATH=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/bin:$PATH\n",
    "export LD_LIBRARY_PATH=/LUSTRE/apps/bioinformatica/ruby2/ruby-2.2.0/lib:$LD_LIBRARY_PATH\n",
    "\n",
    "\n",
    "$TRANSRATE/transrate \\\n",
    "--assembly $ENSAMBLE \\\n",
    "--left $R1 \\\n",
    "--right $R2 \\\n",
    "--threads 8 \\\n",
    "--output gcontigs\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14b6c968",
   "metadata": {},
   "source": [
    "---\n",
    "RSEM\n",
    "---\n",
    "Análisis de expresión diferencial con RSEM, en una de las utilidades de trinity estan las opciones de alineación-estimación de la abundacia y estimación de la abundacia de matrices. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a149be82",
   "metadata": {},
   "source": [
    "##### align_and_estimate_abundance.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e6e975d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -p cicese\n",
    "#SBATCH --job-name=rsem\n",
    "#SBATCH --output=rsem-%j.log\n",
    "#SBATCH --error=rsem-%j.err\n",
    "#SBATCH -N 1\n",
    "#SBATCH --mem=100GB\n",
    "#SBATCH --ntasks-per-node=24\n",
    "#SBATCH -t 6-00:00:00\n",
    "#SBATCH --exclusive\n",
    "\n",
    "#export some soft\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/bowtie2/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/samtools-1.17/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/jellyfish-2.3.0/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/salmon/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/RSEM/bin\n",
    "module load conda-2023\n",
    "module load trinityrnaseq-v2.15.1\n",
    "\n",
    "UTIL=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1\n",
    "\n",
    "cd $SLURM_SUBMIT_DIR\n",
    "$UTIL/util/align_and_estimate_abundance.pl \\\n",
    "--seqType fq \\\n",
    "--samples_file samples.txt \\\n",
    "--transcripts good.Trinity.fasta \\\n",
    "--est_method RSEM \\\n",
    "--aln_method bowtie2 \\\n",
    "--trinity_mode \\\n",
    "--prep_reference \\\n",
    "--output_dir RSEM_outdir \\\n",
    "--thread_count=24 \n",
    "\n",
    "exit 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38dd13cc",
   "metadata": {},
   "source": [
    "##### align_and_estimate_abundance.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ab0b221",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -p cicese\n",
    "#SBATCH --job-name=rsem\n",
    "#SBATCH --output=rsem-%j.log\n",
    "#SBATCH --error=rsem-%j.err\n",
    "#SBATCH -N 1\n",
    "#SBATCH --mem=100GB\n",
    "#SBATCH --ntasks-per-node=24\n",
    "#SBATCH -t 6-00:00:00\n",
    "#SBATCH --exclusive\n",
    "\n",
    "#export some soft\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/bowtie2/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/samtools-1.17/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/jellyfish-2.3.0/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/salmon/bin/\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/RSEM/bin\n",
    "module load conda-2023\n",
    "module load trinityrnaseq-v2.15.1\n",
    "\n",
    "UTIL=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1\n",
    "\n",
    "cd $SLURM_SUBMIT_DIR\n",
    "$UTIL/util/abundance_estimates_to_matrix.pl \\\n",
    "--est_method RSEM \\ #el metodo que se usara puede ser RSEM/Kallisto/Salmon\n",
    "./Silvestres_BLA_C1/RSEM.isoforms.results ../Silvestres_BLA_C2_RSEM.isoforms.results ../Silvestres_BLA_C3/RSEM.isoforms.results \\\n",
    "./Silvestres_BLA_F1/RSEM.isoforms.results ../Silvestres_BLA_F2/RSEM.isoforms.results ../Silvestres_BLA_F3/RSEM.isoforms.results \\\n",
    "./Silvestres_LOL_C1/RSEM.isoforms.results ../Silvestres_LOL_C2/RSEM.isoforms.results ../Silvestres_LOL_C3/RSEM.isoforms.results \\\n",
    "./Silvestres_LOL_F1/RSEM.isoforms.results ../Silvestres_LOL_F2/RSEM.isoforms.results ../Silvestres_LOL_F3/RSEM.isoforms.results \\\n",
    "--name_sample_by_basedir \\ #Nombra la columna de la muestra por nombre del directorio en lugar del archivo predeterminado\n",
    "--gene_trans_map good.Trinity.fasta.gene_trans_map \\ ##el archivo sale del analisis anterior de abundance\n",
    "\n",
    "exit 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4ddcb9d",
   "metadata": {},
   "source": [
    "---\n",
    "Trinotate (Anotación)\n",
    "---\n",
    "Se utilizó una base de datos del Dr.Tripp para no generar una nueva, se buscaron y definieron cada una de las rutas de paqueterías y programas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfd3077b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "### Directivas\n",
    "#SBATCH -p cicese\n",
    "#SBATCH --job-name=Anotacion\n",
    "#SBATCH --ntasks-per-node=8\n",
    "#SBATCH -N 1\n",
    "#SBATCH --mem=20GB\n",
    "#SBATCH --output=anotacion-%j.log\n",
    "#SBATCH --error=anotacion-%j.err\n",
    "#SBATCH -t 06-00:00:00\n",
    "\n",
    "BLAST=/LUSTRE/apps/bioinformatica/ncbi-blast-2.13.0+/bin \n",
    "DBS=/LUSTRE/bioinformatica_data/genomica_funcional/tripp/DATABASES\n",
    "\n",
    "#paqueterias\n",
    "PATH=$PATH://LUSTRE/apps/bioinformatica/Trinotate/PerlLib\n",
    "PATH=$PATH:/LUSTRE/bioinformatica_data/genomica_funcional/apps/Trinotate-Trinotate-v4.0.1\n",
    "PATH=$PATH://LUSTRE/apps/bioinformatica/Trinotate/util/admin\n",
    "UTILS=/LUSTRE/apps/bioinformatica/trinityrnaseq-v2.15.1/util/support_scripts\n",
    "\n",
    "#programas \n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/ncbi-blast-2.13.0+/bin\n",
    "export PATH=$PATH:/LUSTRE/bioinformatica_data/genomica_funcional/apps/infernal-1.1.4/src\n",
    "export PATH=$PATH:/LUSTRE/bioinformatica_data/genomica_funcional/apps/infernal-1.1.4\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/RSEM/bin/samtools-1.3/\n",
    "export PATH=$PATH:/LUSTRE/bioinformatica_data/genomica_funcional/apps/hmmer-3.3.2/src\n",
    "export PATH=$PATH:/LUSTRE/bioinformatica_data/genomica_funcional/tripp/apps/tmhmm-2.0c/bin\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/Trinotate\n",
    "export PATH=$PATH:/LUSTRE/apps/bioinformatica/TransDecoder-v5.7.0\n",
    "\n",
    "echo \"Ejecuntandose la anotacion funcional\"\n",
    "echo \"Fecha de inicio: `date`\"\n",
    "v1=`date`\n",
    "\n",
    "cd ${SLURM_SUBMIT_DIR}\n",
    "\n",
    "echo \"Blastx\"\n",
    "echo \"inicio BlastX: `date`\"\n",
    "\n",
    "#Cambiar de nombre el ensamble para evitar nombres de archivos muy largos\n",
    "\n",
    "mv good.Trinity.fasta Trinity.fasta\n",
    "\n",
    "\n",
    "$BLAST/blastx -query Trinity.fasta -db $DBS/uniprot_sprot.pep -num_threads 8 -max_target_seqs 1 -evalue 1E-05 -outfmt 6 > Blastx_swissprot.outfmt6\n",
    "\n",
    "echo \"Resultados de Blastx se guardaron como BlastX_swissprot.outfmt6\"\n",
    "\n",
    "#relacion de isoformas y genes\n",
    "\n",
    "echo \"generacion de gene_trans_map\"\n",
    "\n",
    "$UTILS/get_Trinity_gene_to_trans_map.pl Trinity.fasta > Trinity.gene_trans_map\n",
    "\n",
    "#identificacion de regiones codificantes\n",
    "\n",
    "echo \"Identificacion de regiones codificantes con transdecoder\"\n",
    "echo \"inicio Transdecoder: `date`\"\n",
    "\n",
    "TransDecoder.LongOrfs -t Trinity.fasta --gene_trans_map Trinity.gene_trans_map > transdecoder.log \n",
    "\n",
    "#Predecir probables ORF codificantes\n",
    "echo \"Predecir probables ORF codificantes: `date`\"\n",
    "\n",
    "TransDecoder.Predict -t Trinity.fasta --cpu 8 >> transdecoder.log \n",
    "\n",
    "#BlastP\n",
    "\n",
    "echo \"Inicio de BlastP: `date`\"\n",
    "\n",
    "$BLAST/blastp -query Trinity.fasta.transdecoder.pep -db $DBS/uniprot_sprot.pep -num_threads 8 -max_target_seqs 1 -outfmt 6 -evalue 1e-5 > Blastp_swissprot.outfmt6\n",
    "\n",
    "echo \"se genero el archivo BlastP_swissprot.outfmt6 `date`\"          \n",
    "#Identificacion de dominios proteicos con HMMSCAN                    \n",
    "hmmscan --cpu 8 --domtblout TrinotatePFAM.out $DBS/Pfam-A.hmm Trinity.fasta.transdecoder.pep\n",
    "\n",
    "echo \"hmmscan `date`\"\n",
    "\n",
    "#Prediccion de senhales peptidicas\n",
    "signalp -f short -n signalp.out Trinity.fasta.transdecoder.pep\n",
    "\n",
    "#Prediccion de dominios transmembranales\n",
    "tmhmm --short < Trinity.fasta.transdecoder.pep > tmhmm.out \n",
    "\n",
    "echo \"tmhmm.out: `date`\"\n",
    "\n",
    "\n",
    "###final e imprimir fecha###\n",
    "echo \"Tiempo de inicio: $v1 \"\n",
    "echo \"Tiempo final: `date`\"\n",
    "\n",
    "exit 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79751e93",
   "metadata": {},
   "source": [
    "###### Para conocer más sobre los comandos y flags [Trinotate](https://github.com/Trinotate/Trinotate.github.io/blob/master/deprecated/index.asciidoc)\n",
    "***Falta el programa SignalP"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
